tests/handlers/test_profile.py

```diff
@@ -327,6 +327,47 @@ def test_avatar_constraint_mime_type(self) -> None:
         )
         self.assertFalse(res)
 
+    @unittest.override_config({"allowed_avatar_mimetypes": ["image/png"]})
+    def test_avatar_fetching_metadata_right_source(self) -> None:
+        """Tests that local and remote file are rightly fetched for metadata
+        when checking for avatar size and mime type"""
+        remote_server = "test:8080"
+        store = self.hs.get_datastores().main
+
+        self.get_success(
+            store.store_local_media(
+                media_id="local",
+                media_type="image/png",
+                media_length=50,
+                time_now_ms=self.clock.time_msec(),
+                upload_name=None,
+                user_id=UserID.from_string("@whatever:test"),
+            )
+        )
+        self.get_success(
+            store.store_cached_remote_media(
+                media_id="remote",
+                media_type="image/png",
+                media_length=50,
+                origin=remote_server,
+                time_now_ms=self.clock.time_msec(),
+                upload_name=None,
+                filesystem_id="remote",
+            )
+        )
```

Comment:
suggest putting this next to the code that is going to do the test (ie, *after* `self.handler.check_avatar_size_and_mime_type("mxc://test/local")`).